docs/stateless.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Stateless IRMA server
+---
+
+For each IRMA session the [IRMA server](irma-server.md) needs to keep track of the [session state](irma-protocol#the-session-state). 
+By default the session state is kept in memory. With version 0.9.0 of the [IRMA server](irma-server.md) it is possible to run the IRMA server in a stateless mode. The session data will then be stored in a Redis datastore. This page explains how the stateless IRMA server works and how to run the IRMA server in stateless mode.
+
+## Overview
+During an IRMA session, the IRMA server acts as a state machine which is described in detail on the [IRMA protocol page](http://localhost:3000/docs/next/irma-protocol#the-session-state). The IRMA session progresses through various states such as `INITIALIZED`, `CONNECTED` and `DONE`. The IRMA server contains the logic when to switch from one state into the other. The state itself can be kept in memory or, alternatively, in a Redis data store. Saving the session state in memory requires no extra setup and is currently the default behaviour of the IRMA server. However, the data in memory cannot be shared between several IRMA servers. This means that you cannot scale the IRMA server horizontally when using this in-memory solution. Next to poor scalability, the in-memory solution is problematic when the IRMA server needs to be restarted or - even worse - when the server crashes. In such cases, the in-memory information of all sessions gets irreversibly lost.
+
+With the new feature of running the IRMA server in stateless mode, the session data gets decoupled from the IRMA server making the server itself stateless. The IRMA server still contains the logic when to switch between states but will retrieve and update the session data from and to a dedicated store. Hence, several IRMA servers can be run in parallel and connect to the same data store. Also, no data is lost when the IRMA server restarts or crashes.
```

Comment:
`gets decoupled from the IRMA server making the server itself stateless` should have a comma: `gets decoupled from the IRMA server, making the server itself stateless`.